src/mkdocs_include_markdown_plugin/event.py

```diff
@@ -123,6 +131,11 @@ def read_file(file_path: str, encoding: str) -> str:
     with open(file_path, encoding=encoding) as f:
         return f.read()
 
+def read_http(target_url: str, encoding: str) -> str:
+    """Read an http location and return its content."""
+    response = requests.get(target_url)
+    # print(response.text)
+    return response.text
```

Comment:
Would you mind to create a statically cached version with a persistent sqlite database of request responses? You can use [`platformdirs`](https://pypi.org/project/platformdirs/) to get a suitable directory per OS and [`diskcache`](https://pypi.org/project/diskcache/) for the database. These dependencies should be optional and the cache can't be used if not installed. The cache time it would be great if can be customized as a setting from the *mkdocs.yml* file.

This idea is what I requested in https://github.com/mondeja/mkdocs-include-markdown-plugin/issues/138#issuecomment-1346758414. I find this neccesary because a lot of HTTPs requests include a lot of delay building. With persistent cache in files, you don't need to make the same request multiple times expeding a lot of times in builds.

Reply:
can you point to an example of a project that uses this?  I do not understand a) how to do this, b) why it is necessary.  I would prefer/suggest a timeout instead of going through the trouble of caching.  I don't imagine people creating sites entirely out of remote md files.  This is really just supplemental information that would be nice to include in an mkdocs site.  And, in time, if someone wants to improve the function to include caching, they may.  I would try to make this as simple as possible and as light and widely adopted (without extra components) as we can.

Reply:
OK, I'll complete this part. Thanks for the feedback. I think that can be made optional when including a `cache` setting.

Reply:
Thank you. I think that is fair. I will work on the plural issue.